GH-8583: Add Java & DSL samples into docs (#8608)
spring-projects/spring-integration: src/reference/asciidoc/transformer.adoc

Nevertheless, the framework does provide some convenient transformers for dealing with XML-based payloads if that is indeed the right choice for your application.
For more information on those transformers, see <<./xml.adoc#xml,XML Support - Dealing with XML Payloads>>.

==== Configuring a Transformer with Java and other DSLs

> For a simple Java & Annotation configuration, the Spring bean POJO method must be marked with a `@Transformer` annotation and the framework calls it when messages are consumed from an input channel:

Reviewer: ```suggestion
For simple Java & Annotation configuration, the Spring bean POJO method must be marked with a `@Transformer` annotation and the framework calls it when messages are consumed from an input channel:
```

====
[source,java]
----
public class SomeService {

    @Transfomer(inputChannel = "transformChannel", outputChannel = "nextServiceChannel")
    public OutputData exampleTransformer(InputData payload) {
        ...
    }

}
----
====

See more information in the <<./configuration.adoc#annotations, Annotation Support>>.

> For Java, Groovy or Kotlin DSL, the `.transform()` operator of an `IntegrationFlow` represents a transformer endpoint:

Reviewer: ```suggestion
For Java, Groovy or Kotlin DSLs, the `.transform()` operator of an `IntegrationFlow` represents a transformer endpoint:
```

===== Object-to-String Transformer

> Because it is fairly common to use the `toString()` representation of an `Object`, Spring Integration provides an `ObjectToStringTransformer` (see also `Transformers` factory) whose output is a `Message` with a String `payload`.

Reviewer: ```suggestion
Because it is fairly common to use the `toString()` representation of an `Object`, Spring Integration provides an `ObjectToStringTransformer` (see also the `Transformers` factory) where the output is a `Message` with a String `payload`.
```

You can also set the `flatten` attribute to false, as follows:

====
[source, java, role="primary"]
.Java DSL
----
@Bean
public IntegrationFlow someFlow() {
    return IntegrationFlow
             .from("directInput")
             .transform(Transformers.toMap(false))
             .channel("output")
             .get();
}
----
[source, xml, role="secondary"]
.XML
----
<int:object-to-map-transformer input-channel="directInput" output-channel="output" flatten="false"/>
----
====

> Spring Integration provides an XML namespace support for Map-to-Object and Java DSL factory, as the following example shows:

Reviewer: ```suggestion
Spring Integration provides XML namespace support for Map-to-Object and the Java DSL factory has the `fromMap()` method, as the following example shows:
```
